### PR TITLE
monotonic counters should use uint64 data type

### PR DESCRIPTION
**Pull request by copperlight**
* There are cases, when working with `MonotonicCounter`s, where it is possible to experience integer overflow, when using the int64 data type. Using a `uint64` data type offers a means of working around this issue. Atlas stores all data as doubles on the backend, and the actual deltas reported to Atlas are expected to be well within that range. Spectatord provides `MonotonicCounter` as a convenient way of avoiding the need to track state and calculate deltas, and as such, should support `uint64`.

* Update the `MonotonicCounter` parameter name to reflect that it is recording a value, because spectatord is responsible for calculating the delta.

* Clean up parameter names and docstrings. Sort function names.

* Run `go mod tidy`, to remove all of the unused dependency references.

* Add a Makefile to ease common operations.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+	@sed -n 's/^##//p' ${MAKEFILE_LIST} | column -t -s ':' |  sed -e 's/^/ /'
+
+## tidy: format code and tidy modfile
+.PHONY: tidy
+tidy:
+	go fmt ./...
+	go mod tidy -v
+
+## test: run all tests
+.PHONY: test
```
Comment (jivimberg) [inline Makefile:16]: Nice! We should add a command that runs golangci-lint to match the [Github Workflow](https://github.com/Netflix/spectator-go/blob/main/.github/workflows/release.yml#L37)

File: spectator/meter/counter.go [hunk 2]
```diff
@@ -34,16 +34,24 @@ func (c *Counter) Increment() {
...
 		var line = fmt.Sprintf("%s:%s:%f", c.meterTypeSymbol, c.id.spectatordId, delta)
 		c.writer.Write(line)
 	}
 }
 
-// Add is to add a specific int64 delta to the current measurement.
-func (c *Counter) Add(delta int64) {
+// AddUint adds an uint64 delta to the current measurement.
+func (c *Counter) AddUint(delta uint64) {
 	if delta > 0 {
```
Comment (jivimberg) [inline spectator/meter/counter.go:55]: Can a UInt ever be < 0?

**Review by jivimberg — CHANGES_REQUESTED** [review]

**Review by sgg — APPROVED** [review]

File: spectator/meter/counter.go (hunk 2 above)
Comment (copperlight) [reply spectator/meter/counter.go]: This check was overlooked. That being said, this PR has been reworked to focus on `MonotonicCounter` alone.

**Review by jivimberg — APPROVED** [review]
LGTM

**Comment by brharrington** [comment]
Note, even a uint64 could overflow. For monotonic counters it is common to assume the only reason a value could go lower is because of an overflow and account for that when computing the delta. For example the [RRD COUNTER](https://oss.oetiker.ch/rrdtool/doc/rrdcreate.en.html#:~:text=a%20RedHat%20share.-,COUNTER,-is%20for%20continuous) type does that.

**Comment by copperlight** [comment]
> Note, even a uint64 could overflow.

Good point, thank you for this! Part of the motivation for this change is to ensure that the overflow happens in the spectatord  MonotonicCounter and not at the interface between user code and spectator-go, if people are using uint64 in their code.

**Pull request merged**